extension-compose/src/main/java/com/mapbox/maps/extension/compose/MapboxMap.kt

```diff
@@ -27,33 +35,74 @@ public fun MapboxMap(
    */
   modifier: Modifier = Modifier,
   /**
-   * Defines the initialisation configurations for a [MapboxMap].
+   * Defines the initialisation configurations factory for a [MapboxMap].
    *
    * It can only be set once and not mutable after the initialisation. Mutating the [MapInitOptions]
    * during recomposition will result in a [IllegalStateException].
    */
-  mapInitOptions: MapInitOptions = MapInitOptions(LocalContext.current.applicationContext),
+  mapInitOptionsFactory: (Context) -> MapInitOptions = { context -> MapInitOptions(context) },
+  /**
+   * Settings for showing the attribution icon on the map.
+   */
+  attributionSettings: AttributionSettings = DefaultSettingsProvider.defaultAttributionSettings(
+    LocalContext.current.applicationContext
+  ),
+  /**
+   * Settings for showing the compass icon on the map.
+   */
+  compassSettings: CompassSettings = DefaultSettingsProvider.defaultCompassSettings(
+    LocalContext.current.applicationContext
+  ),
+  /**
+   * Gesture configuration allows to control the user touch interaction.
+   */
+  gesturesSettings: GesturesSettings = DefaultSettingsProvider.defaultGesturesSettings,
+  /**
+   * Settings for showing a location puck on the map.
+   */
+  locationComponentSettings: LocationComponentSettings = DefaultSettingsProvider.defaultLocationComponentSettings(
+    LocalContext.current.applicationContext,
+  ),
+  /**
+   * Additional settings for showing a location puck on the map.
+   */
+  locationComponentSettings2: LocationComponentSettings2 = DefaultSettingsProvider.defaultLocationComponentSettings2,
+  /**
+   * Settings for showing the Mapbox logo on the map.
+   */
+  logoSettings: LogoSettings = DefaultSettingsProvider.defaultLogoSettings(
+    LocalContext.current.applicationContext
+  ),
+  /**
+   * Settings for showing the scale bar on the map.
+   */
+  scaleBarSettings: ScaleBarSettings = DefaultSettingsProvider.defaultScaleBarSettings(
+    LocalContext.current.applicationContext
+  ),
   /**
    * Callback to be invoked when the user clicks on the map view.
    */
-  onMapClickListener: OnMapClickListener? = null,
+  onMapClickListener: OnMapClickListener = OnMapClickListener { false },
   /**
    * Callback to be invoked when the user long clicks on the map view.
    */
-  onMapLongClickListener: OnMapLongClickListener? = null,
+  onMapLongClickListener: OnMapLongClickListener = OnMapLongClickListener { false },
```

Comment:
@jush I've reverted the listeners to be non-null, according to the official compose [guideline](https://android.googlesource.com/platform/frameworks/support/+/androidx-main/compose/docs/compose-api-guidelines.md#compose-api-design-patterns-default-policies-through-hoisted-state-objects-do)

> Jetpack Compose framework and Library development MUST NOT use null as a sentinel indicating that the composable function should internally remember {} its own state. This can create accidental inconsistent or unexpected behavior if null has a meaningful interpretation for the caller and is provided to the composable function by mistake.

but I guess we can discuss it.

Reply:
TIL.
Makes sense then to not use `null`. There will be a call to the native side per each tap but maybe it's not big deal.

Reply:
@jush I've actually added logic to provide the default static empty listener and compare the listener with the default instance to do no-ops when the default empty listener is used.

Reply:
Indeed, I realized after I commented. I should have read the rest of the code!